src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/StaffRefCreateFunctionalTest.java

```diff
@@ -274,4 +279,75 @@ void createStaffProfile_CwdAdminRoleOnly() {
 
     }
 
+    @Test
+    @ToggleEnable(mapKey = DELETE_CASEWORKER_BY_ID_OR_EMAILPATTERN, withFeature = true)
+    @ExtendWith(FeatureToggleConditionExtension.class)
+    //this test verifies that a User Profile is deleted by ID
+    static void deleteStaffProfileById() {
+        StaffProfileCreationRequest staffProfileCreationRequest = caseWorkerApiClient
+                .createStaffProfileCreationRequest();
+        Response response = caseWorkerApiClient.createStaffUserProfile(staffProfileCreationRequest);
+
+        StaffProfileCreationResponse staffProfileCreationResponse =
+                response.getBody().as(StaffProfileCreationResponse.class);
+
+        String caseWorkerIds = staffProfileCreationResponse.getCaseWorkerId();
+        assertNotNull(caseWorkerIds);
+
+        //delete user
+        caseWorkerApiClient.deleteCaseworkerByIdOrEmailPattern(
+                "/refdata/case-worker/users?userId=" + caseWorkerIds, NO_CONTENT);
+
+        //search for deleted user
+        Response fetchResponse = caseWorkerApiClient.getMultipleAuthHeadersInternal(ROLE_CWD_SYSTEM_USER)
+                .body(UserRequest.builder().userIds(List.of(caseWorkerIds)).build())
+                .post("/refdata/case-worker/users/fetchUsersById/")
+                .andReturn();
+
+        //assert that delete user is not found
+        assertThat(fetchResponse.getStatusCode()).isEqualTo(404);
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {""})
```

Comment:
extract delete endpoint logic to separate method and use that method in this test case and also after all cleanUpTestData method, and you don't need ParameterizedTest and ValueSource